### ISSUE TITLE
Make And Bible work better with e-ink devices

### ISSUE THREAD

**Issue by tuomas2** [AndBible/and-bible#1126]
> Yes, there are E-Ink devices based on Android. I personally use the ONYX 
BOOX Note Pro with Android 9 installed. As I have the AndBible app 
installed on my mobile phone, I am basically already used to the design and 
would love to be able to use it on the E-Ink tablet as well (as this is the 
primary device for my Bible studies). For now, the contrast of the top 
navigation bar is so low, that the icons can't be seen at all. That's by 
now means a complaint, just an idea on how to improve the app or make it 
more "versatile". I really like the app, and I am very thankful for it - it 
is truly a blessing!

https://support.andbible.org/scp/tickets.php?id=561

**Comment by agrogers** [comment]
I've been thinking that being able to choose your own colour for the interface would appeal to some and might make it a more desirable app for those users. And Bible I see more as a tool to get work done and is prepared to sacrifice some aesthetic elements for functionality. That is my preference too. 

But being able to choose a color other than gray as the main interface color might work. That seemed to be what this request was heading towards. So the request could have broader appeal.

**Comment by tuomas2** [comment]
Added strings so mode could be implemented

**Comment by tuomas2** [comment]
Related #282

**Comment by tuomas2** [comment]
Also add And Bible to app stores that are used by e-ink devices (which one(s))?

**Comment by alerque** [comment]
I just came along to report the same issue. I have a Boox Nova Air now, and the same issue as described above. The navigation bar is completely functional if you can guess where to press (as I'm familiar with the UI from my phone this works for me) but the buttons in the header are completely invisible. Something about the gray on gray color scheme ends up with both foreground and background rendered white. A way to tweak the color scheme or a "high contrast" mode should solve this. One caveat would be for first time users finding the option might be difficult on first use. I don't know if there is an Android API to show whether the screen is an e-ink device, but if so defaulting such devices to the better visibility scheme would be useful.

As for app stores, the Boox devices have a custom one with a very limited selection of apps and I have no idea how you would get listed there. It is clearly something they hand-curated with APK's copied from other sources (and mostly out of date), but it was quite simple to load F-Droid and I'm using that as the primary way to get apps. AndBible is already listed there.

**Comment by tuomas2** [comment]
there is no emulator so it's difficult to work on this unless I get a device.

**Comment by alerque** [comment]
I don't think you need a Boox specific emulator, I would guess that ANY Android e-ink devices are going to have the same issue set. Are there *any* e-ink emulators?

**Comment by tuomas2** [comment]
I guess it would be helpful just to have monochrome display emulated (as far as e-ink is monochrome - am I correct?).
I did investigate this and so far I did not find any solution. Let me know if you find a solution. If there is a way to emulate, it helps a lot.

**Comment by tuomas2** [comment]
Perhaps you could consider contributing a PR? Changing some colors is not very complicated nor requires a lot of AB code knowledge.

**Comment by alerque** [comment]
I am traveling and away from development tooling right now, but might try to take a stab at this later in December when I am home again. Its been a while since I built anything for Android, but I should be able to build and sideload onto my device for testing.

The display is monochrome, but it emulates greyscale pretty effectively and to a more limited extent even color with special dithering patterns (similar saturations of red are dithered differently than green for example, one might be dotted and the other stripped). I think part of the problem is the dithering process tries to guess when more contrast is needed for readability, and in the case of the header bar it is deciding that both the foreground (buttons) and background (bar color) need to be (re)moved in the interest of contrast. I have messed with the display contrast settings and it affects a lot of other colors but I can't get it to draw anything in the AndBible header (except the verse reference which comes across fine). I don't know if that is a standard Android accessibility API or something specific to Onyx (the Android firmware varient).

**Comment by agrogers** [comment]
So you think if we just change the background color of the header it might
solve the problem? If so, I can make up a build with a different header
color for you test.

___________________________
*www.TenOutOfTen.org* <https://www.TenOutOfTen.org>
***@***.***
(+61) 0477 174 193 (Australia)
(+95) 09 250018669 (Myanmar)

**Comment by tuomas2** [comment]
Implemented in #3354
